src/SoftRobots/component/constraint/model/CableModel.h

```diff
@@ -158,15 +185,22 @@ class SOFA_SOFTROBOTS_API CableModel : virtual public SoftRobotsConstraint<DataT
 
     void checkIndicesRegardingState();
     void initActuatedPoints();
+    void initCableActionAreas();
+    void computePointsActionArea();
+    unsigned int computeClosestIndice(const Coord& position);
+    void getPositionFromTopology(Coord& position, const int& index);
+    SReal getDistanceToTriangle(const Coord& position, const Triangle& triangle, Coord& projectionOnTriangle);
+    void computeBarycentric(const Triangle& triangle, const Coord& p, Real& alpha, Real& beta);
 
     void drawPullPoint(const VisualParams* vparams);
     void drawPoints(const VisualParams* vparams);
     void drawLinesBetweenPoints(const VisualParams* vparams);
+    void drawPulledAreas(const VisualParams* vparams);
 };
 
 // Declares template as extern to avoid the code generation of the template for
 // each compilation unit. see: http://www.stroustrup.com/C++11FAQ.html#extern-templates
-extern template class CableModel<defaulttype::Vec3Types>;
+//extern template class CableModel<defaulttype::Vec3Types>;
```

Comment:
Why the comment?

This is how I would do it:

```suggestion
if !defined(SOFTROBOTS_COMPONENT_BEHAVIOR_CONSTRAINT_MODEL_CABLEMODEL_CPP)
extern template class SOFA_SOFTROBOTS_API CableModel<defaulttype::Vec3Types>;
extern template class SOFA_SOFTROBOTS_API CableModel<defaulttype::Vec2Types>;
#endif
```

and in the cpp file, add the following line just before the first include:
```cpp
#define SOFTROBOTS_COMPONENT_BEHAVIOR_CONSTRAINT_MODEL_CABLEMODEL_CPP
```